oi/type_graph/TopoSorter.cpp

```diff
@@ -54,30 +54,68 @@ void TopoSorter::visit(Type& type) {
   type.accept(*this);
 }
 
-void TopoSorter::visit(Class& c) {
-  for (const auto& param : c.templateParams) {
-    visit(param.type);
+void TopoSorter::visitAfter(Type& type) {
+  typesToSort_.push(type);
+}
+
+void TopoSorter::visitAfter(Type* type) {
+  if (type) {
+    visitAfter(*type);
   }
+}
+
+void TopoSorter::visit(Class& c) {
   for (const auto& parent : c.parents) {
     visit(*parent.type);
   }
   for (const auto& mem : c.members) {
     visit(*mem.type);
   }
+  for (const auto& param : c.templateParams) {
+    visit(param.type);
+  }
   sortedTypes_.push_back(c);
 
-  // Same as pointers, child do not create a dependency so are delayed until the
-  // end
+  // Same as pointers, children do not create a dependency so are delayed until
+  // the end TODO comment
```

Comment:
Finish this off!

Reply:
Looks great and I couldn't see anything TBH.

One general point. Don't be afraid to put comments in so that anyone coming into this code can readily pick this up. It's a small thing but I do like at least one comment that gives an overview of what the code in a file is trying to do without the reader having to ingest it all. Not for now but just to bare in mind.

Reply:
There is a short overview comment in the header file: https://github.com/facebookexperimental/object-introspection/blob/5560624c0b5301f8b5fb0eb955bffc0560f8e384/oi/type_graph/TopoSorter.h#L29

The `visitAfter` function is probably the most confusing bit (as it's not part of a standard topological sort) and is conviently the part I only half wrote a comment for 😅